src/runtime/src/core/iterator.rs

```diff
@@ -244,13 +244,11 @@ pub fn make_module() -> ValueMap {
                 match iter_output {
                     Ok(Output::Value(value)) => {
                         result = Some(match result {
-                            Some(result) => {
-                                match vm.run_binary_op(Operator::Less, result, value) {
-                                    Ok(Bool(true)) => value,
-                                    Ok(Bool(false)) => result,
-                                    Err(error) => return Err(error.with_prefix("iterator.max")),
-                                }
-                            }
+                            Some(result) => match vm.run_binary_op(Operator::Less, result, value) {
+                                Ok(Bool(true)) => value,
+                                Ok(Bool(false)) => result,
```

Comment:
Errors like: 
```rust
  Ok(unexpected) => {
    return Err(external_error!(
      "iterator.max: Expected Bool from < comparison, found '{}'", type_as_string(&unexpected)
    );
  },
```
will be needed in these functions, I missed that in the WIP commit.

Reply:
OK, I'll change it.